shared/templates/create_permissions.py

```diff
@@ -0,0 +1,217 @@
+
+        dir_path, file_name, uid, gid, mode = path_info[0:5]
+
+        # build a string out of the path that is suitable for use in id tags
+        # example:	/etc/resolv.conf --> _etc_resolv_conf
+        # path_id maps to FILEID in the template
+
+        if len(path_info) == 6:
+            path_id = path_info[5]
+        elif file_name == '[NULL]':
+            path_id = re.sub('[-\./]', '_', dir_path)
+        elif re.match(r'\^.*\$', file_name, 0):
+            path_id = re.sub('[-\./]', '_', dir_path) + '_' + re.sub('[-\\\./^$*(){}|]',
+                                                                     '_', file_name)
+            # cleaning trailing end multiple underscores, make sure id is lowercase
+            path_id = re.sub(r'_+','_', path_id)
+            path_id = re.sub(r'_$','', path_id)
+            path_id = path_id.lower()
+        else:
+            path_id = re.sub('[-\./]', '_', dir_path) + '_' + re.sub('[-\./]',
+                                                                     '_', file_name)
+            path_id = path_id.lower()
+
+        # build a string that contains the full path to the file
+        # full_path maps to FILEPATH in the template
+        if file_name == '[NULL]':
+            full_path = dir_path
+        else:
+            full_path = dir_path + '/' + file_name
+
+        if target == "bash":
+            if not re.match(r'\^.*\$', file_name, 0):
+                if mode:
+                    self.file_from_template(
+                        "./template_BASH_file_permissions",
+                        {
+                            "FILEPATH":      full_path,
+                            "FILEMODE":      mode,
+                        },
+                        "./bash/file_permissions{0}.sh", path_id
+                    )
+                if uid:
+                    self.file_from_template(
+                        "./template_BASH_file_owner",
+                        {
+                            "FILEPATH":      full_path,
+                            "FILEUID":       uid,
+                        },
+                        "./bash/file_owner{0}.sh", path_id
+                    )
+                if gid:
+                    self.file_from_template(
+                        "./template_BASH_file_groupowner",
+                        {
+                            "FILEPATH":      full_path,
+                            "FILEGID":       gid,
+                        },
+                        "./bash/file_groupowner{0}.sh", path_id
+                    )
+                self.file_from_template(
+                    "./template_BASH_permissions",
+                    {
+                        "FILEPATH":      full_path,
+                        "FILEMODE":      mode,
+                        "FILEUID":       uid,
+                        "FILEGID":       gid,
+                    },
+                    "./bash/permissions{0}.sh", path_id
```

Comment:
I believe this bash remediation should only be generated if `mode`, `uid` and `gid` are defined.

Reply:
I updated this by adding a line at the top of [`create_permissions.py`](https://github.com/cipherboy/scap-security-guide/blob/banach-space/shared/templates/create_permissions.py#L17) and adding gated checks in the jinja templates.